File: thisboot/src/main/java/com/brageast/blog/thisboot/config/MongodbConfig.java
package com.brageast.blog.thisboot.config;

import com.mongodb.MongoClientOptions;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoClients;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.mongo.MongoProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.mongodb.config.AbstractReactiveMongoConfiguration;
import org.springframework.data.mongodb.repository.config.EnableReactiveMongoRepositories;

@Slf4j
@Configuration
//@EnableReactiveMongoRepositories
public class MongodbConfig extends AbstractReactiveMongoConfiguration {

    private final MongoProperties mongoProperties;

    @Autowired
    public MongodbConfig(MongoProperties mongoProperties) {
        this.mongoProperties = mongoProperties;
    }

    @Bean
    public MongoClientOptions mongoOptions() {
        return MongoClientOptions
                .builder()
                .maxConnectionIdleTime(60000)
                .build();
    }

    @Override
    public MongoClient reactiveMongoClient() {
        return MongoClients.create();
    }

    @Override
    protected String getDatabaseName() {
        return mongoProperties.getDatabase();
    }
}

File: thisboot/src/main/kotlin/com/brageast/blog/thisboot/config/MongodbConfig.kt
package com.brageast.blog.thisboot.config

import com.mongodb.reactivestreams.client.MongoClient
import org.springframework.boot.autoconfigure.mongo.MongoProperties
import org.springframework.context.annotation.Configuration
import org.springframework.data.mongodb.config.AbstractReactiveMongoConfiguration
import org.springframework.data.mongodb.config.EnableMongoAuditing
import org.springframework.data.mongodb.repository.config.EnableReactiveMongoRepositories

@Configuration
@EnableMongoAuditing
@EnableReactiveMongoRepositories(basePackages = ["com.brageast.blog.thisboot.repository"])
class MongodbConfig(
        val mongoProperties: MongoProperties,
        val mongoClient: MongoClient
) : AbstractReactiveMongoConfiguration() {

    override fun reactiveMongoClient(): MongoClient = mongoClient

    override fun getDatabaseName(): String = mongoProperties.database

    override fun autoIndexCreation(): Boolean = false

}